packages/pdf-viewer/VerticalPages.tsx

```diff
@@ -45,15 +85,47 @@ export default function VerticalPages(props: VerticalPagesProps) {
 				resizeTimer = null;
 			}
 		};
-	}, [props.pdf]);
+	}, []);
+
+
+	useEffect(() => {
+		let scrollTimer: number = null;
+
+		const saveScroll = () => {
+			const scrollTop = props.container.current.scrollTop;
+			if (props.rememberScroll && props.pdfId) {
+				sessionStorage.setItem(`pdf.${props.pdfId}.scrollTop`, `${scrollTop}`);
+			}
+		};
+
+		const onScroll = () => {
+			if (scrollTimer) {
+				clearTimeout(scrollTimer);
+				scrollTimer = null;
+			}
+			scrollTimer = window.setTimeout(saveScroll, 200);
+		};
+
+		props.container.current.addEventListener('scroll', onScroll);
+
+		return () => {
+			props.container.current.removeEventListener('scroll', onScroll);
+			if (scrollTimer) {
+				clearTimeout(scrollTimer);
+				scrollTimer = null;
+			}
+		};
+
+	}, []);
```

Comment:
No dependencies listed here but I can clearly see some - props.rememberScroll, props.pdfId

Your components will be full of rendering bugs if you're not careful.